src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesProjectTreeProvider.cs

```diff
@@ -0,0 +1,973 @@
+                                (treeSnapshot, configuredProjectExports, cancellationToken) =>
+                                    {
+                                        var dependenciesNode = CreateDependenciesFolder(null);                                        
+                                        dependenciesNode = CreateOrUpdateSubTreeProviderNodes(dependenciesNode, 
+                                                                                              cancellationToken);
+
+                                        return Task.FromResult(new TreeUpdateResult(dependenciesNode, true));
+                                    });                            
+                        }
+
+                    },
+                    registerFaultHandler: true);
+            }
+        }
+
+        /// <summary>
+        /// Disposes this instance.
+        /// </summary>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                foreach (var provider in SubTreeProviders)
+                {
+                    provider.Value.DependenciesChanged -= OnDependenciesChanged;
+                }
+
+                ProjectContextUnloaded?.Invoke(this, new ProjectContextEventArgs(this));
+            }
+
+            base.Dispose(disposing);
+        }
+
+        private void OnDependenciesChanged(object sender, DependenciesChangedEventArgs e)
+        {
+            var nowait = SubmitTreeUpdateAsync(
+                (treeSnapshot, configuredProjectExports, cancellationToken) =>
+                {
+                    var dependenciesNode = treeSnapshot.Value.Tree;
+                    dependenciesNode = CreateOrUpdateSubTreeProviderNode(dependenciesNode,
+                                                                         e.Provider,
+                                                                         changes: e.Changes,
+                                                                         cancellationToken: cancellationToken,
+                                                                         catalogs: e.Catalogs);
+
+                    ProjectContextChanged?.Invoke(this, new ProjectContextEventArgs(this));
+
+                    // Note: this is legacy code that changes data source version, assuming this is correct
+                    // We're excluding this data source for now until we figure out how the 
+                    // PhysicalProjectTreeProvider can reconcile grafts with different ActiveProjectConfiguration
+                    // versions. With the below line we had bugs like 535352 where the product would flag an 
+                    // internal error because this version would "decrement" across an active project config change.
+                    //var dataSources = e.DataSourceVersions
+                    //                   .Add(ProjectTreeDataSources.ReferencesFolderProjectSnapshotVersion, 0L);
```

Comment:
Is this comment still valid? If there is any further follow up here please file a bug to track it.